File: 1301-1400/1343-Number of Sub-arrays of Size K and Average Greater than or Equal to Threshold/1343-Number of Sub-arrays of Size K and Average Greater than or Equal to Threshold.cpp
class Solution {
public:
    int numOfSubarrays(vector<int>& arr, int k, int threshold) {
        int count = 0, sum = 0;
        for (int i = 0; i < arr.size(); ++i) {
            sum += arr[i];
            if (i >= k) {
                sum -= arr[i - k];
            }
            if (i >= k - 1 && sum / k >= threshold) {
                ++count;
            }
        }
        
        return count;
    }
};

File: 1301-1400/1343-Number of Sub-arrays of Size K and Average Greater than or Equal to Threshold/1343-Number of Sub-arrays of Size K and Average Greater than or Equal to Threshold.py
class Solution:
    def numOfSubarrays(self, arr: List[int], k: int, threshold: int) -> int:
        count = s = 0
        for i, a in enumerate(arr):
            s += a
            if i >= k:
                s -= arr[i - k]
            if i >= k - 1 and s / k >= threshold:
                count += 1
        return count
